pkg/asset/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,125 @@
+package tls
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"html/template"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/tls"
+)
+
+const (
+	KubeconfigUserNameAdmin   = "admin"
+	KubeconfigUserNamekubelet = "kubelet"
+)
+
+var kubeconfigTemplate = `
```

Comment:
any reason why not use `https://godoc.org/k8s.io/client-go/tools/clientcmd/api/v1#Config` directly? Why does it have to be a template.

Reply:
here's an example in tnc https://github.com/coreos-inc/tectonic-operators/blob/e750236e76e8166bf1bd0493aff6ff519c9a3560/controller/node/pkg/tnc/tnc.go#L582